io_gripper_controller/src/io_gripper_controller.cpp

```diff
@@ -0,0 +1,1369 @@
+          get_node()->get_logger(),
+          "Size of configuration map parameter cannot be zero if configurations are defined.");
+        return CallbackReturn::FAILURE;
+      }
+    }
+    else
+    {
+      RCLCPP_FATAL(
+        get_node()->get_logger(),
+        "Configuration joints have to be defined if configurations are provided.");
+      return CallbackReturn::FAILURE;
+    }
+  }
+
+  // gripper_specific_sensors parameter
+  if (!params_.gripper_specific_sensors.empty())
+  {
+    // sensors interfaces parameter
+    if (params_.sensors_interfaces.gripper_specific_sensors_map.empty())
+    {
+      RCLCPP_FATAL(
+        get_node()->get_logger(),
+        "Size of sensors interfaces parameter cannot be zero if gripper specific sensors are "
+        "defined.");
+      return CallbackReturn::FAILURE;
+    }
+    else
+    {
+      // if sensor input string is empty then return failure
+      for (const auto & [key, val] : params_.sensors_interfaces.gripper_specific_sensors_map)
+      {
+        if (val.input == "")
+        {
+          RCLCPP_FATAL(
+            get_node()->get_logger(),
+            "Size of sensor input string parameter cannot be empty ("
+            ").");
+          return CallbackReturn::FAILURE;
+        }
+      }
+    }
+  }
+  return CallbackReturn::SUCCESS;
+}
+
+void IOGripperController::prepare_command_and_state_ios()
+{
+  auto parse_interfaces_from_params = [](
+                                        const std::vector<std::string> & parameter_values,
+                                        const double & value,
+                                        std::unordered_map<std::string, double> & ios,
+                                        std::unordered_set<std::string> & interface_list)
+  {
+    for (const auto & itf : parameter_values)
+    {
+      if (!itf.empty())
+      {
+        ios[itf] = value;
+        interface_list.insert(itf);
+      }
+    }
+  };
+
+  // make full command ios lists -- just once
+  parse_interfaces_from_params(
+    params_.open.set_before_command.high, 1.0, open_ios_.set_before_command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.set_before_command.low, 0.0, open_ios_.set_before_command_ios, command_if_ios);
+
+  parse_interfaces_from_params(
+    params_.open.command.high, 1.0, open_ios_.command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.command.low, 0.0, open_ios_.command_ios, command_if_ios);
+
+  parse_interfaces_from_params(
+    params_.open.set_after_command.high, 1.0, open_ios_.set_after_command_ios, command_if_ios);
+  parse_interfaces_from_params(
+    params_.open.set_after_command.low, 0.0, open_ios_.set_after_command_ios, command_if_ios);
+
+  for (const auto & key : params_.close.set_before_command.high)
+  {
+    if (!key.empty())
+    {
+      set_before_command_close.push_back(key);
+      set_before_command_close_values.push_back(1.0);
+      command_if_ios.insert(key);
+    }
+  }
```

Comment:
This should be adjusted in the same way as open_ios.

```suggestion
  parse_interfaces_from_params(
    params_.close.set_before_command.high, 1.0, close_ios_.set_before_command_ios, command_if_ios);
```

And do this for all command below. Don't forget to change the usage of the variables in the rest of the code in the same way I did for the `open_ios_`.